framework/Models/Code.py

```diff
@@ -308,8 +312,11 @@ def createNewInput(self,currentInput,samplerType,**kwargs):
     if not os.path.exists(subDirectory):
       os.mkdir(subDirectory)
     for index in range(len(newInputSet)):
-      newInputSet[index].setPath(subDirectory)
-      shutil.copy(self.oriInputFiles[index].getAbsFile(),subDirectory)
+      subSubDirectory = os.path.join(subDirectory,newInputSet[index].subDirectory)
+      if newInputSet[index].subDirectory.strip() != "" and not os.path.exists(subSubDirectory):
+        os.mkdir(subSubDirectory)
```

Comment:
So, it looks like from the [code coverage](http://mooseframework.org/docs/PRs/285/python-coverage/_home_moosetest_civet_client_root_0_raven_framework_Models_Code_py.html) the directory creation in `initialize` is never executed. Do you anticipate this to change, or should we leave this in?